engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EjeaRunningJobSpec.scala

```diff
@@ -64,30 +64,64 @@ class EjeaRunningJobSpec extends EngineJobExecutionActorSpec with Eventually wit
+          ejea.stateName should be(RunningJob)
+          ejea ! failureNonRetryableResponse
+          expectCacheWriteForFailedNonRetryableJob(failureNonRetryableResponse, SuccessfulCallCacheHashes)
+        }
+
+        s"Handle receiving FailedNonRetryable then HashError correctly in $mode mode" in {
+          ejea = ejeaInRunningState(mode)
+          ejea ! failureNonRetryableResponse
+          eventually { ejea.stateData should be(FailedNonRetryableResponseData(failureNonRetryableResponse, None)) }
+          ejea.stateName should be(RunningJob)
+          ejea ! hashError
+          helper.jobStoreProbe.expectMsgPF(max = awaitTimeout, hint = "Job Store Write") {
+            case RegisterJobCompleted(jobKey, JobResultFailure(returnCode, reason, isRetryable)) =>
+              validateJobStoreKey(jobKey)
+              returnCode should be(failedRc)
+              reason should be(failureReason)
+              isRetryable should be(false)
+          }
+        }
+
+        s"Handle receiving HashError then FailedNonRetryable correctly in $mode mode" in {
+          ejea = ejeaInRunningState(mode)
+          ejea ! hashError
+          eventually { ejea.stateData should be(ResponsePendingData(helper.backendJobDescriptor, helper.bjeaProps, Some(Failure(hashError.reason)))) }
+          ejea.stateName should be(RunningJob)
+          ejea ! failureNonRetryableResponse
+          helper.jobStoreProbe.expectMsgPF(max = awaitTimeout, hint = "Job Store Write") {
+            case RegisterJobCompleted(jobKey, JobResultFailure(returnCode, reason, isRetryable)) =>
+              validateJobStoreKey(jobKey)
+              returnCode should be(failedRc)
+              reason should be(failureReason)
+              isRetryable should be(false)
+          }
+        }
       }
     }
 
-    // What's with all this maker stuff? You can't use the "helper" outside of spec tests so we can't instantiate these responses
-    // yet. So, we take the functions instead and call them when we need them
-    List(("FailedRetryableResponse", failureRetryableResponse _, true),
-         ("FailedNonRetryableResponse", failureNonRetryableResponse _, false)
-    ) foreach { case (name, responseMaker, retryable) =>
-      s"register '$name's with the JobStore" in {
-        ejea = ejeaInRunningState()
-        val response = responseMaker.apply
-        ejea.underlyingActor.receive.isDefinedAt(response) should be(true)
-        ejea ! response
-
-        helper.jobStoreProbe.expectMsgPF(max = awaitTimeout, hint = "Job Store Write") {
-          case RegisterJobCompleted(jobKey, JobResultFailure(returnCode, reason, isRetryable)) =>
-            validateJobStoreKey(jobKey)
-            returnCode should be(failedRc)
-            reason should be(failureReason)
-            isRetryable should be(retryable)
-        }
+    s"register 'FailedRetryableResponse's with the JobStore" in {
```

Comment:
Yes, this test ensure that we go into "UpadtingJobStore" mode when getting a `FailedNonRetryableResponse`, which we don't do anymore since we go into "write to cache" mode.